extensions/http-provisioner/src/main/java/org/eclipse/dataspaceconnector/transfer/provision/http/AbstractHttpResourceDefinition.java

```diff
@@ -0,0 +1,60 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.transfer.provision.http;
+
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.ResourceDefinition;
+
+import java.util.Objects;
+
+/**
+ * Defines common attributes for HTTP-based provisioning asset resources.
+ */
+public abstract class AbstractHttpResourceDefinition extends ResourceDefinition {
+    protected String dataAddressType;
+
+    public String getDataAddressType() {
```

Comment:
i think this should be documented, i.e. what does it refer to (i.e. that it defines what  `DataAddress.getType()`s this provisioner handles)